### PR TITLE
Support Zone and Graph sentences

### PR DESCRIPTION
**Pull request by darionyaphet**

### CODE REVIEW

File: src/planner/PlanNode.h [hunk 1]
```diff
@@ -106,12 +106,25 @@ class PlanNode {
         kShowCharset,
         kShowCollation,
         kShowConfigs,
+        kShowGroups,
```
Comment (Shylock-Hg) [inline src/planner/PlanNode.h:109]: Better to put into `zone related`

File: src/planner/PlanNode.cpp [hunk 2]
```diff
@@ -181,6 +181,10 @@ const char* PlanNode::toString(PlanNode::Kind kind) {
             return "ShowCollation";
         case Kind::kShowConfigs:
             return "ShowConfigs";
+        case Kind::kShowGroups:
```
Comment (Shylock-Hg) [inline src/planner/PlanNode.cpp:184]: ditto.

File: src/planner/PlanNode.h (hunk 1 above)
Comment (darionyaphet) [reply src/planner/PlanNode.h:109]: ok  fixed

**Comment by darionyaphet** [comment]
any one would review this one ?

File: src/parser/AdminSentences.cpp [hunk 3]
```diff
@@ -80,6 +88,20 @@ std::string SpaceOptList::toString() const {
...
 
+std::string ZoneList::toString() const {
+    std::string buf;
+    buf.reserve(256);
+    for (auto &zone : zones_) {
+        buf += std::to_string(zone);
+        buf += ",";
+    }
+    if (!buf.empty()) {
+        buf.resize(buf.size() - 1);
```
Comment (CPWstatic) [inline src/parser/AdminSentences.cpp:99]: `buf.pop_back()`

File: src/parser/parser.yy [hunk 4]
```diff
@@ -401,8 +411,13 @@ unreserved_keyword
     | KW_SHORTEST           { $$ = new std::string("shortest"); }
     | KW_COUNT_DISTINCT     { $$ = new std::string("count_distinct"); }
     | KW_CONTAINS           { $$ = new std::string("contains"); }
+    | KW_NOT_CONTAINS       { $$ = new std::string("not contains"); }
     | KW_STARTS             { $$ = new std::string("starts"); }
+    | KW_STARTS_WITH        { $$ = new std::string("starts with"); }
+    | KW_NOT_STARTS_WITH    { $$ = new std::string("not starts with"); }
     | KW_ENDS               { $$ = new std::string("ends"); }
+    | KW_ENDS_WITH          { $$ = new std::string("ends with"); }
+    | KW_NOT_ENDS_WITH      { $$ = new std::string("not ends with"); }
```
Comment (CPWstatic) [inline src/parser/parser.yy:420]: We don't have to handle these.

File: src/parser/parser.yy [hunk 5]
```diff
@@ -2269,16 +2360,52 @@ show_config_item
...
         auto sentence = new CreateSpaceSentence($4, $3);
         sentence->setOpts($6);
         $$ = sentence;
     }
+    // | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN KW_ON zone_list {
+    //     auto sentence = new CreateSpaceSentence($4, $3);
+    //     sentence->setOpts($6);
+    //     sentence->setZones($9);
+    //     $$ = sentence;
+    // }
```
Comment (CPWstatic) [inline src/parser/parser.yy:2408]: Need some comment for the unimplemented features.

File: src/util/ZoneUtil.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+#define UTIL_ZONEUTIL_H_
+
+#include "common/base/Base.h"
+#include "common/base/StatusOr.h"
+#include "parser/MaintainSentences.h"
+
+namespace nebula {
+namespace graph {
+
+class ZoneUtil final {
```
Comment (CPWstatic) [inline src/util/ZoneUtil.h:17]: What's the plan to do with the `ZoneUtil` ans `GroupUtil`?

File: src/parser/parser.yy (hunk 4 above)
Comment (darionyaphet) [reply src/parser/parser.yy:420]: but when I commit code it report some warning message .

File: src/parser/parser.yy (hunk 5 above)
Comment (darionyaphet) [reply src/parser/parser.yy:2408]: yep  I will remove it and support later

File: src/util/ZoneUtil.h (hunk 6 above)
Comment (darionyaphet) [reply src/util/ZoneUtil.h:17]: I  will move some zone and group assign code into it.

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Review by CPWstatic — APPROVED** [review]

File: src/parser/parser.yy [hunk 7]
```diff
@@ -2282,6 +2373,31 @@ show_config_item
...
+//         $$->addZone($1);
+//     }
+//     | zone_list COMMA zone {
+//         $$ = $1;
+//         $$->addZone($3);
+//     }
+//     | zone_list COMMA {
+//         $$ = $1;
+//     }
+//     ;
```
Comment (dutor) [inline src/parser/parser.yy:2388]: Will these still be useful?

**Review by dutor — COMMENTED** [review]
Great work!

File: src/parser/parser.yy (hunk 7 above)
Comment (darionyaphet) [reply src/parser/parser.yy:2388]: sorry it should be cleaning

**Review by dutor — APPROVED** [review]
LGTM.

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**